### ISSUE TITLE
[HOLD for payment] [$1000] Bug: Mobile app still shows workspace after removed from Web reported by @aimane-chnaif 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11807]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Create workspace if not exists
2. Refresh web and mobile apps and confirm that workspace shows
3. Close mobile app
4. Delete workspace on web
5. Open mobile app

## Expected Result:
Workspace should be deleted on mobile app as well

## Actual Result:
Deleted workspace still remaining on mobile app

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.14
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/195644451-d78c48fd-823b-413f-ab46-b282928eb102.mp4

https://user-images.githubusercontent.com/43996225/195644586-577c054a-4007-4adb-9310-3f2e13ef1e1e.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @aimane-chnaif 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665065560778309

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

## Upwork Automation - Do Not Edit
- Upwork Job URL: https://www.upwork.com/jobs/~0174f3f0c9ecb3ddeb
- Upwork Job ID: 1594940834258837504

**Comment by zanyrenney** [comment]
Ugh yeah this is weird, I don't know why the disconnect between mobile and web here - will get some eng eyes!

**Comment by madmax330** [comment]
Yeah they should be synced, not sure why mobile isn't receiving the updates. Maybe we're not sending the updates for this action via Pusher? 
I think this should be internal

**Comment by MonilBhavsar** [comment]
cc @aldo-expensify If you have any idea about this issue or can take a look

**Comment by aldo-expensify** [comment]
We are sending the pusher events to remove the workspace from the workspace owner here: https://github.com/Expensify/Web-Expensify/blob/1a494780e09845a37da08f631f67b81fabcc1bf5/lib/PolicyAPI.php#L4137-L4143

If you see the workspace being removed in the same device (web) that took the action, that is the pusher update event removing it. I don't know why the mobile device is missing it, but sounds like it could be missing pusher events when the App is closed and maybe then there is not request to update the policy data when it is open? This may be not noticeable in chats conversation because I think we send a request to get the latests chats once you come back (not 100% sure about this).

**Comment by aldo-expensify** [comment]
I would say a first step would be to make sure that pusher events don't get lost if the App is closed. It makes sense to me that they get lost in this case. If that is the case, I feel like the way around it is to bring the policy data again when you open the App.

**Comment by aldo-expensify** [comment]
Considering what I said, I think this can be `External`

**Comment by isabelastisser** [comment]
Was it agreed this is an internal or external issue? What are the next steps I need to take here? Thanks!

**Comment by aldo-expensify** [comment]
I haven't marked this a `external` because I haven't investigated further to confirm what I posted above. If we want to refetch this data once the App reopens, it may require a change in the Web-API.

I'll assign it to me to investigate it when I have some time and let you know.

**Comment by aldo-expensify** [comment]
Tried to reproduce in staging using my Android phone + web and it worked just fine. In the video they seem to be using a simulator, maybe it doesn't work well in that case?

**Comment by aldo-expensify** [comment]
@kavimuru has this been reproducing using a real device (not a simulator)?

**Comment by aldo-expensify** [comment]
@aimane-chnaif 👋 since you reported the bug, have you seen this happening in real devices?

**Comment by aldo-expensify** [comment]
Sorry, I still can't reproduce. After I open the Android app, I do see the workspace active (not deleted), but one or two seconds after it changes to deleted and the chats get archived. I'm guessing there is a delay because the app is fetching some data?

**Comment by aldo-expensify** [comment]
Some additional info... 

- I checked and each time we re open the app, we send a request with command `OpenApp` which will pull the current policies (workspaces)
- The policies brought by `OpenApp` of course don't include the deleted workspace (there isn't a key setting the workspace to null).

I guess if we miss the pusher event that remove the workspace. The Onyx updates coming from `OpenApp` are not going to remove the deleted workspace. The only way to clear it at this point would be to sign out and sign in.

**Comment by aldo-expensify** [comment]
ahhhhh was getting confused by the workspace disappearing from the LHN and the chats archived. The chats get properly archived because the OpenApp updates them, but the workspace was staying and I didn't notice because I had to go to the settings.

This is also reproducible with two browser sessions:

1. Login in new dot in two browser sessions with the same account
2. Create a workspace
3. Make sure both sessions see the workspace
4. On one session close the tab, but do not close the window (session). The idea is that you will reopen the tab later, and this is just for missing the pusher event when we delete the workspace.
5. On the session that remained open, delete the workspace
6. Reopen the New Expensify tab in the session you closed it... you will see that the chats are archived, but if you go to settings, you will see the deleted Workspace there 🐛

**Comment by aldo-expensify** [comment]
cc @luacmartins 

I found [this code](https://github.com/Expensify/Web-Expensify/blob/2e88c1b42cc09c25a24075bf1addf8461229d95b/lib/AppInit.php#L60-L65) in the API that is supposed to use the policy ids in the client to respond if any of the policies should be removed

The problem is that in the App, when we call `OpenApp` here, `policyIDList` has not been initialized yet 

https://github.com/Expensify/App/blob/071f51ccd3f651e138ac11d2b00684fcb37d9fe6/src/libs/actions/App.js#L119

**Comment by aldo-expensify** [comment]
I didn't have time this week to get back to this, but I have a PR that I think should solve it. I'll test it early next week and finish it.

**Comment by aldo-expensify** [comment]
I'm working on this, but got blocked by a bug in the backend: https://github.com/Expensify/App/issues/12330

I'll finish the PR https://github.com/Expensify/App/pull/12129 and put in on hold on the backend fix: https://github.com/Expensify/Web-Expensify/pull/35386

**Comment by aldo-expensify** [comment]
No overdue, the backend PR was merged and the front end PR is ready for review: https://github.com/Expensify/App/pull/12129

**Comment by JmillsExpensify** [comment]
Would you like some BZ assist on this issue? Let me know and I'll add myself.

**Comment by JmillsExpensify** [comment]
Still working through the PR.

**Comment by parasharrajat** [comment]
@JmillsExpensify Could you please assign me here as C+ for internal PR review? Thanks

**Comment by JmillsExpensify** [comment]
Done! Thanks

**Comment by JmillsExpensify** [comment]
Linked PR is merged, so I'm adding the external label briefly to get an Upwork job created and handle Rajat's payment when this is deployed.

**Comment by JmillsExpensify** [comment]
@aimane-chnaif just returning from holidays. Yes definitely!

**Comment by JmillsExpensify** [comment]
In fact, the linked PR was merged a couple of days ago, so it's time to make sure all contributors are hired and we can issue payout. Just hired @parasharrajat and then @aimane-chnaif please apply to the job as well for reporting.

**Comment by JmillsExpensify** [comment]
@parasharrajat @aimane-chnaif Please apply to the [Upwork job](https://www.upwork.com/jobs/~0174f3f0c9ecb3ddeb) so that we can close this issue out. 

 - Contributor+: @parasharrajat $1,000
 - Issue reporter: @aimane-chnaif $250

**Comment by parasharrajat** [comment]
Done.

**Comment by JmillsExpensify** [comment]
All contractors paid out in Upwork.

**Comment by JmillsExpensify** [comment]
Regression testing issue linked above, and PRs are on production so I'm closing this issue out.
